assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js

```diff
@@ -88,45 +104,59 @@ const Idea = ( { postEditURL, name, text, topics, buttons } ) => {
 					<p className="googlesitekit-idea-hub__idea--text">{ text }</p>
 				</Cell>
 				<Cell smSize={ 4 } mdSize={ 3 } lgSize={ 3 } className="googlesitekit-idea-hub__idea--actions">
-					{ buttons.includes( IDEA_HUB_BUTTON_DELETE ) && (
-						<Button
-							onClick={ handleDelete }
-							icon={ <DeleteIcon /> }
-							className="googlesitekit-idea-hub__actions--delete"
-							disabled={ isProcessing }
```

Comment:
@danielgent let's do the following:
* Add the `ACTIVITY_IS_PROCESSING` activity
* Update `handleDelete`, `handlePin`, and `handleUnpin` callbacks to set the `ACTIVITY_IS_PROCESSING` before starting an operation and to remove the activity at the end of the callback.
* Add back `disabled` attribute to buttons and use `ACTIVITY_IS_PROCESSING === activity` condition as its value to disable buttons when we are handling a certain operation.

Reply:
Makes sense! Done (I think)